test/cases/api.sh

```diff
@@ -198,73 +325,104 @@ do
-chmod 400 ./keypair.pem
-
-# Create an instance based on the ami
-$AWS_CMD ec2 run-instances --image-id "$AMI_IMAGE_ID" --count 1 --instance-type t2.micro --key-name "key-for-$AMI_IMAGE_ID" > "$WORKDIR/instances.json"
-INSTANCE_ID=$(jq -r '.Instances[].InstanceId' "$WORKDIR/instances.json")
-
-$AWS_CMD ec2 wait instance-running --instance-ids "$INSTANCE_ID"
-
-$AWS_CMD ec2 describe-instances --instance-ids "$INSTANCE_ID" > "$WORKDIR/instances.json"
-HOST=$(jq -r '.Reservations[].Instances[].PublicIpAddress' "$WORKDIR/instances.json")
-
-echo "⏱ Waiting for AWS instance to respond to ssh"
-for LOOP_COUNTER in {0..30}; do
-    if ssh-keyscan "$HOST" > /dev/null 2>&1; then
-        echo "SSH is up!"
-        # ssh-keyscan "$PUBLIC_IP" | sudo tee -a /root/.ssh/known_hosts
-        break
-    fi
-    echo "Retrying in 5 seconds... $LOOP_COUNTER"
-    sleep 5
-done
+  fi
+
+
+  # Create an instance based on the ami
+  $AWS_CMD ec2 run-instances --image-id "$AMI_IMAGE_ID" --count 1 --instance-type t2.micro --key-name "key-for-$AMI_IMAGE_ID" > "$WORKDIR/instances.json"
+  AWS_INSTANCE_ID=$(jq -r '.Instances[].InstanceId' "$WORKDIR/instances.json")
+
+  $AWS_CMD ec2 wait instance-running --instance-ids "$AWS_INSTANCE_ID"
 
-# Check if postgres is installed
-ssh -oStrictHostKeyChecking=no -i ./keypair.pem "$SSH_USER"@"$HOST" rpm -q postgresql
+  $AWS_CMD ec2 describe-instances --instance-ids "$AWS_INSTANCE_ID" > "$WORKDIR/instances.json"
+  HOST=$(jq -r '.Reservations[].Instances[].PublicIpAddress' "$WORKDIR/instances.json")
 
-if [ "$SHARE_OK" != 1 ]; then
+  echo "⏱ Waiting for AWS instance to respond to ssh"
+  for LOOP_COUNTER in {0..30}; do
+      if ssh-keyscan "$HOST" > /dev/null 2>&1; then
+          echo "SSH is up!"
+          # ssh-keyscan "$PUBLIC_IP" | sudo tee -a /root/.ssh/known_hosts
+          break
+      fi
+      echo "Retrying in 5 seconds... $LOOP_COUNTER"
+      sleep 5
+  done
+
+  # Check if postgres is installed
+  ssh -oStrictHostKeyChecking=no -i ./keypair.pem "$SSH_USER"@"$HOST" rpm -q postgresql
+
+  if [ "$SHARE_OK" != 1 ]; then
     echo "EC2 snapshot wasn't shared with the AWS_API_TEST_SHARE_ACCOUNT. 😢"
     exit 1
-fi
+  fi
+}
+
+# Verify image in Compute Node on GCP
+function verifyInGCP() {
+  # Authenticate
+  $GCP_CMD auth activate-service-account --key-file "$GOOGLE_APPLICATION_CREDENTIALS"
+  # Extract and set the default project to be used for commands
+  GCP_PROJECT=$(jq -r '.project_id' "$GOOGLE_APPLICATION_CREDENTIALS")
+  $GCP_CMD config set project "$GCP_PROJECT"
+
+  # Verify that the image was shared
+  SHARE_OK=1
+  $GCP_CMD compute images get-iam-policy "$GCP_IMAGE_NAME" > "$WORKDIR/image-iam-policy.json"
+  SHARED_ACCOUNT=$(jq -r '.bindings[0].members[0]' "$WORKDIR/image-iam-policy.json")
+  SHARED_ROLE=$(jq -r '.bindings[0].role' "$WORKDIR/image-iam-policy.json")
+  if [ "$SHARED_ACCOUNT" != "$GCP_API_TEST_SHARE_ACCOUNT" ] || [ "$SHARED_ROLE" != "roles/compute.imageUser" ]; then
+    SHARE_OK=0
+  fi
+
```

Comment:
This should probably also instantiate / boot the image like for AWS, right? For a follow-up?

Reply:
Yes, this should be definitely added eventually. @Gundersanne also said that this should be good enough to start with. My expectation is that this may not be trivial, because GCP currently can not inject SSH keys into the instance which uses our imported image. This may require introducing the GCP image type.

Reply:
Makes sense.